torchinductor/codegen/triton.py

```diff
@@ -591,8 +591,11 @@ def load(self, name: str, index: sympy.Expr, upcast: bool = False):
         if upcast:
             line += ".to(tl.float32)"
 
-        if self.inside_reduction and "rmask" not in mask:
+        if self.inside_reduction and "rmask" not in mask and self.loads != self.compute:
```

Comment:
```suggestion
        if self.inside_reduction and "rmask" not in mask and self._load_mask is None:
```

Might be more readable and future-proof.  Should do the same thing.

Reply:
Hmm, I tried your suggestion locally, but it doesn't solve the problem. I guess `self._load_mask` is not a good indictor of whether it is an indirect-load? I can add another flag if that's more readable.

```
tmp0 = tl.load(in_ptr0 + r0, rmask)
tmp2 = tl.load(in_ptr2 + tmp0, None).to(tl.float32)
```

Reply:
Ah you are be right.  Let's land this version then.